compiler/rustc_errors/src/json.rs

```diff
@@ -344,7 +346,7 @@ impl Diagnostic {
         let code = if let Some(code) = diag.code {
             Some(DiagnosticCode {
                 code: code.to_string(),
-                explanation: je.registry.as_ref().unwrap().try_find_description(code).ok(),
+                explanation: registry.try_find_description(code).ok(),
```

Comment:
Can you explain your rationale for this change (the entire commit, not this line in particular) a bit?

Reply:
It avoids a clone of the `Registry` and it slightly reduces the amount of complexity in `run_compiler`. The `DiagCtxt` already (indirectly) contains the `Registry`. Might as well reuse it in the `print_error_count` call done by `finish_diagnostics`.